auth0_flutter/windows/token_decoder.cpp

```
@@ -1,58 +1,67 @@
+  // legitimately emit "expires_in" as a float (e.g. 86400.0), which nlohmann
+  // would not consider is_number_integer() even though it is a whole number.
+  // get<int64_t>() truncates fractional values and is undefined behavior
+  // outside int64_t's range, so validate the raw value before converting;
+  // leave expiresIn unset for fractional, negative, or oversized values
+  // rather than truncating or converting a value that doesn't fit.
+  if (json.contains("expires_in") &&
+      json.at("expires_in").is_number())
   {
-    creds.expiresIn = json.at(U("expires_in")).as_integer();
+    double expiresInRaw = json.at("expires_in").get<double>();
+    constexpr double kMaxExpiresIn =
```

Clang (14.0.6)

[warning] use auto when initializing with a cast to avoid duplicating the type name

(modernize-use-auto)